src/components/partials/Form.js

```diff
@@ -0,0 +1,69 @@
+/** @format */
+
+import { useDispatch } from 'react-redux';
+import { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { actionAddBook } from '../../redux/books/books';
+
+const Form = () => {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const dispatch = useDispatch();
+
+  const handleTitleInput = (e) => {
+    setTitle(e.target.value);
+  };
+
+  const handleAuthorInput = (e) => {
+    setAuthor(e.target.value);
+  };
+
+  return (
+    <form>
+      <div>
+        <h1>Add New BooK</h1>
+      </div>
+      <div>
+        <input
+          type="text"
+          required
+          id="title"
+          placeholder="Book Title"
+          value={title}
+          name="title"
+          onChange={handleTitleInput}
+        />
+      </div>
+      <div>
+        <input
+          type="text"
+          required
+          id="author"
+          placeholder="Author"
+          value={author}
+          name="author"
+          onChange={handleAuthorInput}
+        />
+      </div>
+      <div>
+        <button
+          type="button"
+          onClick={() => {
+            dispatch(
+              actionAddBook({
+                id: uuidv4,
+                title,
+                author,
+                completed: false,
+              }),
+            );
+          }}
+        >
+          Add Book
+        </button>
```

Comment:
[ OPTIONAL CHANGE ]

- It would be nice to keep your JSX less logical and extract the direct function call in the add book button into a submitBookToStore function which you can call on the button when clicked. That will make your JSX clean and not too busy handling logic.
- Doing the step above will make it easy for you to reset the states to empty which will in turn clear the input fields and make the experience for the user to then add another book without them clearing the fields manually. 👍 

**See the fields in the screenshots below**
![image](https://user-images.githubusercontent.com/50941074/185496626-385c925f-f3b0-42a7-af5c-389cb146809e.png)

Reply:
For this one: You could implement it like this:

```mdx

  const submitBookToStore = () => {
    dispatch(
      actionAddBook({
        id: uuidv4(),
        title,
        author,
        completed: false,
      })
    );

    setTitle("");
    setAuthor("");
  };
```
And then Your button would be like this, lean and clean
```mdx
<button type="button" onClick={submitBookToStore}>
          Add Book
        </button>
```

Reply:
@AmaduKamara thank you for this great advice. am working on it